src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3677,13 +3677,48 @@ public override Type MakeArrayType(int rank)
         #region Invoke Member
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern bool CanValueSpecialCast(RuntimeType valueType, RuntimeType targetType);
+        private static extern bool CanPrimitiveWiden(CorElementType valueType, CorElementType targetType);
```

Comment:
We have two managed implementations of CanPrimitiveWiden in CoreLib:

https://github.com/dotnet/runtime/blob/c20bdf6de579d42faa4d019c8332edbeef618c44/src/libraries/System.Private.CoreLib/src/System/Reflection/InvokeUtils.cs#L274
https://github.com/dotnet/runtime/blob/c20bdf6de579d42faa4d019c8332edbeef618c44/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs#L131

Use one of those?

Reply:
Can dedup those while at it https://github.com/am11/runtime/commit/d046fcbde539d0285c5ca4e91e6300f12b30b4fe.

Reply:
Can dedup those while at it https://github.com/am11/runtime/commit/d046fcbde539d0285c5ca4e91e6300f12b30b4fe. (local RO `primitiveWidenTable` is a bit better than the shared since it's unused elsewhere)

Reply:
Nice.